src/pyscaffoldext/custom_extension/extension.py

```diff
@@ -232,6 +233,29 @@ def add_doc_requirements(struct: Structure, opts: ScaffoldOpts) -> ActionParams:
     return merge(struct, files), opts
 
 
+def add_cirrus_ci(
+    cirrus_actions: List[Action], struct: Structure, opts: ScaffoldOpts
+) -> ActionParams:
+    """Opportunistically add CirrusCI config if no other CI service was added."""
+
+    uses_github_actions = struct.get(".github", {}).get("workflows") is not None
+    other_ci_files = [".gitlab-ci.yml"]
```

Comment:
This is a bit add-hoc, although completely fine if we just want to consider extensions that we have completely control over (no 3rd-party).